### PR TITLE
NIFI-11266 PutGoogleDrive, ListGoogleDrive, FetchGoogleDrive can't ac…

### PR DESCRIPTION
**Pull request by krisztina-zsihovszki**
…cess a SharedDrive

# Summary

[NIFI-11266 ](https://issues.apache.org/jira/browse/NIFI-11266)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 11
  - [x] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by mark-bathori** [comment]
Reviewing

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/ListGoogleDrive.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ public void onScheduled(final ProcessContext context) throws IOException {
 
         HttpTransport httpTransport = new ProxyAwareTransportFactory(proxyConfiguration).create();
 
-        driveService = createDriveService(context, httpTransport, DriveScopes.DRIVE_METADATA_READONLY);
```
Comment (mark-bathori) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/ListGoogleDrive.java:197]: I think this change is unnecessary. The `DRIVE_METADATA_READONLY` scope should be enough for the List processor.

**Review by mark-bathori — COMMENTED** [review]
Thanks @krisztina-zsihovszki for fixing this issue. I have a minor comment on the changes, other then that looks good to me.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/PutGoogleDrive.java [hunk 2]
```diff
@@ -187,7 +187,7 @@ public class PutGoogleDrive extends AbstractProcessor implements GoogleDriveTrai
             REL_FAILURE
     )));
 
-    public static final String MULTIPART_UPLOAD_URL = "https://www.googleapis.com/upload/drive/v3/files?uploadType=multipart";
+    public static final String MULTIPART_UPLOAD_URL = "https://www.googleapis.com/upload/drive/v3/files?uploadType=multipart&supportsAllDrives=true";
```
Comment (nandorsoma) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/PutGoogleDrive.java:190]: This doesn't seem to be right that we need to include that flag while `.setSupportsAllDrives(true)` is already set on the `driveRequest`. Could you open a ticket for the Drive API team to investigate it?

**Review by nandorsoma — APPROVED** [review]
Thanks for your contribution @krisztina-zsihovszki and your review @mark-bathori. The PR looks good, I tested it and worked as expected.

Will let Github Actions complete and then can merge.

**Pull request closed without merging**